src/main/java/com/redislabs/redisgraph/RedisGraphAPI.java

```diff
@@ -96,13 +109,179 @@ public String deleteGraph() {
 		}
     }
 
+
+    /**
+     * Sends command
+     * @param conn
+     * @param provider
+     * @param args
+     * @return
+     */
     private BinaryClient sendCommand(Jedis conn, ProtocolCommand provider, String ...args) {
         BinaryClient binaryClient = conn.getClient();
         binaryClient.sendCommand(provider, args);
         return binaryClient;
     }
+
+
+    /**
+     * Sends the command with --COMPACT flag
+     * @param conn
+     * @param provider
+     * @param args
+     * @return
+     */
+    private BinaryClient sendCompactCommand(Jedis conn, ProtocolCommand provider, String ...args) {
```

Comment:
private BinaryClient sendCompactCommand(Jedis conn, ProtocolCommand provider, String ...args) {
        String[] t = new String[args.length +1];
        System.arraycopy(args, 0 , t, 0, args.length);
        t[args.length]="--COMPACT";
        return this.sendCommand(conn, provider, t);
    }